tensorpack/train/base.py

```diff
@@ -53,6 +55,21 @@ def __init__(self, config):
         self.sess = tf.Session(config=self.config.session_config)
         self.coord = tf.train.Coordinator()
 
+    def apply_tfslim_collections(self, loss_op):
+        # add regularization loss
+        regul_loss = ops.get_collection(ops.GraphKeys.REGULARIZATION_LOSSES)
+        if len(regul_loss) > 0:
+            loss_op += tf.add_n(regul_loss)
+
+        # add non-gradient updates, like batch-norm
+        update_ops = set(ops.get_collection(ops.GraphKeys.UPDATE_OPS))
+        if update_ops:
+            with ops.control_dependencies(update_ops):
+                barrier = control_flow_ops.no_op(name='update_barrier')
+            loss_op = control_flow_ops.with_dependencies([barrier], loss_op)
```

Comment:
This is from
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/slim/python/slim/learning.py#L428-L430

I think it is just a nice way to name this barrier for TensorBoard.